src/Bottle/instance-provider.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Register an instance provider inside a generic provider.
+ *
+ * @param {String} name - The name of the service
+ * @param {Function} Factory - The factory function, matches the signature required for the
+ * `factory` method
+ * @return Bottle
+ */
+var instanceProvider = function instanceProvider(name, Factory) {
+  return provider.call(this, name, function GenericProvider() {
```

Comment:
@nzack I think you can simplify this by doing something like the following:

```js
var instanceProvider = function instanceProvider(name, Factory) {
  return factory.call(this, name, function GenericInstanceFactory(container) {
      return {
        instance : Factory.bind(null, container)
      };
  });
};
```

Then you can drop the new function constructer below.